lib/compress.js

```diff
@@ -2893,7 +2920,7 @@ merge(Compressor.prototype, {
                 });
             }
         }
-        return self;
+        return self.evaluate(compressor)[0];
```

Comment:
@alexlamsl You can keep those new tests but the `sub_properties` tests you've added would still produce the same result without this PR. I'm looking for AST_Sub cases with `-c unsafe=false` that produce different results with and without this PR. I could have sworn that I had created such a case, but I can't seem to recall or manufacture one now.